src/lib.rs

```diff
@@ -254,3 +265,334 @@ impl<'a, NP: ?Sized + NixPath>  NixPath for Option<&'a NP> {
         }
     }
 }
+
+/*
+ *
+ * ===== Null terminated slices for exec =====
+ *
+ */
+
+use std::ops::{Deref, DerefMut};
+use std::mem::transmute;
+use std::{iter, io};
+
+/// An error returned from the [`TerminatedSlice::from_slice`] family of
+/// functions when the provided data is not terminated by `None`.
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub struct NotTerminatedError {
+    // private initializers only
+    _inner: (),
+}
+
+impl NotTerminatedError {
+    fn not_terminated() -> Self {
+        NotTerminatedError {
+            _inner: (),
+        }
+    }
+}
+
+impl error::Error for NotTerminatedError {
+    fn description(&self) -> &str { "data not terminated by None" }
+}
+
+impl fmt::Display for NotTerminatedError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", error::Error::description(self))
+    }
+}
+
+impl From<NotTerminatedError> for io::Error {
+    fn from(e: NotTerminatedError) -> io::Error {
+        io::Error::new(io::ErrorKind::InvalidInput,
+                       error::Error::description(&e))
+    }
+}
+
+/// An error returned from [`TerminatedVec::from_vec`] when the provided data is
+/// not terminated by `None`.
+#[derive(Clone, PartialEq, Eq)]
```

Comment:
Missing `Copy` and `Debug`.

Reply:
Debug is defined below this (it's manually implemented because we want to impl it regardless of whether `T: Debug` or not), and `Copy` cannot be derived as it contains a `Vec`.

Reply:
Whoops, sorry I didn't read that closely enough!